### PR TITLE
feat: Scrollbar enhancement

### PR DESCRIPTION
**Pull request by tinaClin**
image

image

image

image

### CODE REVIEW

File: packages/styled-ui-docs/pages/scrollbar.mdx [hunk 1]
```diff
@@ -10,6 +10,64 @@ import {
...
+<Stack spacing="4x">
+  <Box>
+    <Text size="xl" marginBottom="2x">auto</Text>
+    <Scrollbar
+      height={120}
+      border={1}
+      borderColor="#424242"
+      scrollbarVisibility="auto"
+    >
+      <Lorem count={12} />
```
Comment (cheton) [inline packages/styled-ui-docs/pages/scrollbar.mdx]: Adding padding to each example

**Review by cheton — CHANGES_REQUESTED** [review]

**Comment by cheton** [comment]
@tinaClin 

I would suggest removing the `autoHideDelay` prop and use `setTimeout(fn, 0)` in current release.

In the future, we will introduce new transition props (i.e. transition easing, duration, and delay) in a separate PR when #297 is merged.

@tinaClin removed in [883ae6c]

File: packages/styled-ui-docs/pages/scrollbar.mdx [hunk 2]
```diff
@@ -68,6 +68,98 @@ Use the `disabled` attribute, the overflow is clipped, and the rest of the conte
...
 ```
 
+### The minimal thumb size
+
+Use the `minThumbSize` attribute to set the minimal size (in pixels) for thumbs.
+
+```jsx
+function MinThumbSizeScroll() {
+  const [minThumbSize, setMinThumbSize] = React.useState('32');
+  const [dataRow, setDataRow] = React.useState('8');
```
Comment (cheton) [inline packages/styled-ui-docs/pages/scrollbar.mdx]: Can `minThumbSize` and `dataRow` be changed to numeric values?

File: packages/styled-ui-docs/pages/scrollbar.mdx [hunk 3]
```diff
@@ -68,6 +68,98 @@ Use the `disabled` attribute, the overflow is clipped, and the rest of the conte
...
+
+  return (
+    <>
+      <Box marginBottom="2x">
+        <Text marginBottom="1x">The minimal thumb size:</Text>
+        <RadioGroup onChange={e => setMinThumbSize(e.target.value)} value={minThumbSize}>
```
Comment (cheton) [inline packages/styled-ui-docs/pages/scrollbar.mdx]: `setMinThumbSize(Number(e.target.value))`

**Comment by cheton** [comment]
d0b31b1

https://app.circleci.com/pipelines/github/trendmicro-frontend/styled-ui/227/workflows/4c3c1fc9-e1b9-4d2c-978f-3e302117abf7/jobs/385

```
$ lerna exec -- yarn pre-push
lerna notice cli v3.22.1
lerna info versioning independent
lerna info ci enabled
lerna info Executing command in 3 packages: "yarn pre-push"
$ eslint src
$ eslint --ext .js,.jsx .

/home/circleci/project/packages/react-styled-ui/src/Scrollbar/Scrollbar.js
  104:73  error  Operator '?' must be spaced                                                                                         space-infix-ops
  120:6   error  Unnecessary semicolon                                                                                               no-extra-semi
  383:6   error  React Hook useEffect has a missing dependency: 'update'. Either include it or remove the dependency array           react-hooks/exhaustive-deps
  389:6   error  React Hook useEffect has a missing dependency: 'handleDragStart'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
  418:47  error  Trailing spaces not allowed                                                                                         no-trailing-spaces
  445:2   error  Missing semicolon                                                                                                   semi
  449:2   error  Missing semicolon                                                                                                   semi
  453:2   error  Missing semicolon                                                                                                   semi
  457:2   error  Missing semicolon                                                                                                   semi
  461:2   error  Missing semicolon                                                                                                   semi

/home/circleci/project/packages/react-styled-ui/src/Scrollbar/styles.js
  44:15  error  Strings must use singlequote  quotes

✖ 11 problems (11 errors, 0 warnings)
  9 errors and 0 warnings potentially fixable with the `--fix` option.
```



@tinaClin fixed in [643eb9c]

File: packages/styled-ui-docs/pages/scrollbar.mdx (hunk 2 above)
Comment (tinaClin) [reply packages/styled-ui-docs/pages/scrollbar.mdx]: @tinaClin fixed in  [2d67361]

File: packages/styled-ui-docs/pages/scrollbar.mdx (hunk 3 above)
Comment (tinaClin) [reply packages/styled-ui-docs/pages/scrollbar.mdx]: @tinaClin fixed in  [2d67361]

File: packages/styled-ui-docs/pages/scrollbar.mdx (hunk 1 above)
Comment (tinaClin) [reply packages/styled-ui-docs/pages/scrollbar.mdx]: Fixed in [ad786d5]

**Review by cheton — APPROVED** [review]

File: packages/react-styled-ui/src/Scrollbar/Scrollbar.js [hunk 4]
```diff
@@ -35,31 +41,35 @@ const Scrollbar = forwardRef((
...
+  const prevPageXRef = useRef(0);
+  const prevPageYRef = useRef(0);
 
   const viewRef = useRef(null);
   const trackHorizontalRef = useRef(null);
   const trackVerticalRef = useRef(null);
   const thumbHorizontalRef = useRef(null);
   const thumbVerticalRef = useRef(null);
 
+  const scrollbarWidth = getScrollbarWidth();
```
Comment (cheton) [inline packages/react-styled-ui/src/Scrollbar/Scrollbar.js]: @tinaClin 
Need to refactor the code or leave a comment here to mention that `getScrollbarWidth` will access to DOM element, which is not safe for SSR. It's better to calculate the scrollbar width after the first render, or just hide the scrollbar for the first render, using the `useEffect` Hook.

Comment (tinaClin) [reply packages/react-styled-ui/src/Scrollbar/Scrollbar.js]: Thanks @cheton! Fixed in [58793a3]

**Pull request merged**